File: Hard/2026-06-17-3614-Process-String-with-Special-Operations-II/solution.js
var processString = function(s, k) {
    let result = [];
    let length = 0;

    for (let char of s) {
        if (char >= 'a' && char <= 'z') {
            result.push(char);
            length++;
        } else if (char === '*') {
            if (length > 0) {
                result.pop();
                length--;
            }
        } else if (char === '#') {
            if (length > 0) {
                let current = result.join('');
                result.push(...current.split(''));
                length *= 2;
            }
        } else if (char === '%') {
            result.reverse();
        }
    }

    return k < length ? result[k] : '.';
};

File: Hard/2026-06-17-3614-Process-String-with-Special-Operations-II/solution.py
class Solution:
    def processString(self, s: str, k: int) -> str:
        result = []
        length = 0
        
        for char in s:
            if char.islower():
                result.append(char)
                length += 1
            elif char == '*':
                if length > 0:
                    result.pop()
                    length -= 1
            elif char == '#':
                result.extend(result)
                length *= 2
            elif char == '%':
                result.reverse()
        
        if k >= length:
            return '.'
        return result[k]